docs/token-builder.md

```diff
@@ -0,0 +1,288 @@
+# Token Builder API
+
+## Overview
+
+The Token Builder is comprised of two parts: the `RequestBuilder` class and a variety of `Token` type classes.
+
+Each of the token classes has its own set of validations which are run when the token is instantiated. Any errors are added to an `errors` property on that instance of the token class.
+
+In `sorcerer`, the Token Parser will extract the tokens from request body and replace any occurrence of the token in an API data source's `url`, `body`, or `headers` with the token's computed value.
+
+## Token Types
+
+The Token Builder API includes several token utility classes, each serving a unique purpose.
+
+Currently supported tokens are:
+
+- ReplaceToken
```

Comment:
suggestion (non-blocking): [linking to the same section](https://stackoverflow.com/a/16426829) can be a good way to shortcut to the relevant part of the doc.